essencium-backend/src/main/java/de/frachtwerk/essencium/backend/security/oauth2/OAuth2SuccessHandler.java

```diff
@@ -179,6 +187,8 @@ private UserInfoEssentials extractUserInfo(
       OAuth2AuthenticationToken authentication, String providerName)
       throws UserEssentialsException {
     final UserInfoEssentials userInfo = new UserInfoEssentials();
+    OAuth2ClientRegistrationProperties.ClientProvider clientProvider =
```

Comment:
Can the result of oAuth2ClientRegistrationProperties.getProvider().get(providerName); be null? From my point of view, if the result isnull an Exception should be thrown right after the call